drone/src/cert/mod.rs

```diff
@@ -62,21 +62,25 @@ pub async fn get_certificate(
     }
 
     builder.terms_of_service_agreed(true);
-    let account = builder.build().await?;
+    let account = builder.build().await.context("Building account")?;
 
     let mut builder = OrderBuilder::new(account);
     builder.add_dns_identifier(format!("*.{}", cluster_domain));
-    let order = builder.build().await?;
+    let order = builder.build().await.context("Building order")?;
 
-    let authorizations = order.authorizations().await?;
+    let authorizations = order
+        .authorizations()
+        .await
+        .context("Fetching authorizations")?;
     for auth in authorizations {
         tracing::info!("Requesting challenge.");
         let challenge = auth
             .get_challenge(DNS_01)
-            .ok_or_else(|| anyhow!("Couldn't obtain dns-01 challenge."))?;
+            .context("Obtaining dns-01 challenge")?;
```

Comment:
why is the .context better than having an ok_or_else here?

Reply:
ok_or_else replaces the original error; context is additive